lib/console/gateways/gateways.ex

```diff
@@ -53,10 +54,35 @@ defmodule Console.Gateways do
       {:error, %Ecto.Changeset{}}
 
   """
-  def create_gateway(attrs \\ %{}) do
-    %Gateway{}
-    |> Gateway.changeset(attrs)
-    |> Repo.insert()
+  def create_gateway(attrs \\ %{}, attempt \\ 1) do
```

Comment:
moved the token generation to HardwareIdentifier.ex 

The reattempts can't be moved because we are using ecto multi, so the unique token failure doesn't trigger until Ecto.Multi.run in this file is reached.

Reply:
ok that's fine